javalib/src/main/scala/java/lang/Throwables.scala

```diff
@@ -163,6 +240,25 @@ class Throwable(s: String, private var e: Throwable)
     duplicates
   }
 
+  def setStackTrace(stackTrace: Array[StackTraceElement]): Unit = {
+
+    if (stackTrace eq null) {
+      throw new java.lang.NullPointerException()
+    }
+
+    for (i <- 0 until stackTrace.length) {
```

Comment:
Please keep the original loop with `while` and `var` here, once again so that `Throwable` does not rely on the Scala library.

Reply:
`while` loop restored.    I understand now why that is needed.   

Guess that the Scala Coursera courses were pretty effective in discouraging use 
of `var` '-)  I understand this is a different context.

I probably have `for` loops and other Scala Libaryisms scattered over all the
javalib PRs I have open. I will keep my eye out for them (and try my best to not
write them in new code).